cogito/protocol.go

```diff
@@ -239,14 +247,33 @@ func (src *Source) Validate() error {
 	sinks := sets.From(src.Sinks...)
 	if sinks.Size() == 0 || sinks.Contains("github") {
 		// Cogito commit Github status mandatory fields.
+		// cannot specify both github_app and access_token
+		if !src.GitHubApp.IsZero() && src.AccessToken != "" {
+			return fmt.Errorf("source: cannot specify both github_app and access_token")
+		}
+
+		// either access_token or github_app must be specified
+		if src.AccessToken == "" && src.GitHubApp.IsZero() {
+			return fmt.Errorf("source: please specify either access_token or github_app")
+		}
+
 		if src.Owner == "" {
 			mandatory = append(mandatory, "owner")
 		}
 		if src.Repo == "" {
 			mandatory = append(mandatory, "repo")
 		}
-		if src.AccessToken == "" {
-			mandatory = append(mandatory, "access_token")
+
+		if !src.GitHubApp.IsZero() {
```

Comment:
so if we are here, either access_tolen or github_app is set; but we still need to check if all github_app mandatory fields are correctly configured if github_app is used...